src/api/CMakeLists.txt

```diff
@@ -23,7 +23,7 @@ set(TSAPI_PUBLIC_HEADERS ${PROJECT_SOURCE_DIR}/include/ts/ts.h ${PROJECT_SOURCE_
                          ${PROJECT_SOURCE_DIR}/include/ts/TsException.h ${PROJECT_BINARY_DIR}/include/ts/apidefs.h
 )
 
-target_link_libraries(tsapi PRIVATE libswoc::libswoc yaml-cpp::yaml-cpp PCRE::PCRE PkgConfig::PCRE2 OpenSSL::SSL)
+target_link_libraries(tsapi PRIVATE libswoc::libswoc yaml-cpp::yaml-cpp OpenSSL::SSL PCRE::PCRE PkgConfig::PCRE2)
```

Comment:
The link order appears arbitrary and I think someone not thinking about it wouldn't realize its importance (this PR to fix it exists, after all). Could we add a comment so someone doesn't accidentally cause a regression?

Reply:
It turned out that there are several places potentially problematic. I'm not sure where would be a good place to put the comment on. I updated the documentation for plugin development for now.

Reply:
I think it'd still be good to have a comment here since this is in core and not covered by the plugin documentation.

What about the plugins that link `OpenSSL::Crypto`? I suspect those should also be updated.

Reply:
> What about the plugins that link OpenSSL::Crypto? I suspect those should also be updated.

Good point. I updated some more places for it. Thanks.

And I added the comment here. Better than not having any.